test/integration_test/marketplace_test.go

```diff
@@ -4,25 +4,71 @@
 package integrationtest
 
 import (
+	"fmt"
+	"strings"
 	"testing"
+	"time"
 
-	"github.com/stretchr/testify/require"
-	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
-
+	"github.com/hashicorp/go-version"
 	corev1 "github.com/libopenstorage/operator/pkg/apis/core/v1"
+	testutil "github.com/libopenstorage/operator/pkg/util/test"
 	"github.com/libopenstorage/operator/test/integration_test/types"
 	ci_utils "github.com/libopenstorage/operator/test/integration_test/utils"
+	"github.com/sirupsen/logrus"
+	"github.com/stretchr/testify/require"
 )
 
 var marketplaceOperatorTestCases = []types.TestCase{
 	{
 		TestName:        "BasicInstallViaOcpMarketplace",
 		TestrailCaseIDs: []string{""},
-		TestSpec: ci_utils.CreateStorageClusterTestSpecFunc(&corev1.StorageCluster{
-			ObjectMeta: metav1.ObjectMeta{Name: "test-stc"},
-		}),
+		TestSpec: func(t *testing.T) interface{} {
+			// Create PX vSphere secret if provisioning PX on vSphere based OCP
+			if ci_utils.IsOcp && ci_utils.CloudProvider == "vsphere" {
+				if ci_utils.PxVsphereUsername == "" || ci_utils.PxVspherePassword == "" {
+					require.NoError(t, fmt.Errorf("Unable to create Portworx vSPhere secret because --portworx-vsphere-username or --portworx-vsphere-password are not passed"))
+				}
+				err := ci_utils.CreatePxVsphereSecret(ci_utils.PxNamespace, ci_utils.PxVsphereUsername, ci_utils.PxVspherePassword)
+				require.NoError(t, err)
+			}
+			// Construct StorageCluster
+			cluster := &corev1.StorageCluster{}
+			cluster.Name = "test-stc"
+			err := ci_utils.ConstructStorageCluster(cluster, ci_utils.PxSpecGenURL, ci_utils.PxSpecImages)
+			require.NoError(t, err)
+			return cluster
+		},
 		TestFunc: BasicInstallViaOcpMarketplace,
 	},
+	{
+		TestName:        "BasicUpgradeOperatorViaOcpMarketplace",
+		TestrailCaseIDs: []string{""},
+		TestSpec: func(t *testing.T) interface{} {
+			// Create PX vSphere secret if provisioning PX on vSphere based OCP
+			if ci_utils.IsOcp && ci_utils.CloudProvider == "vsphere" {
+				if ci_utils.PxVsphereUsername == "" || ci_utils.PxVspherePassword == "" {
+					require.NoError(t, fmt.Errorf("Unable to create Portworx vSPhere secret because --portworx-vsphere-username or --portworx-vsphere-password are not passed"))
+				}
+				err := ci_utils.CreatePxVsphereSecret(ci_utils.PxNamespace, ci_utils.PxVsphereUsername, ci_utils.PxVspherePassword)
+				require.NoError(t, err)
+			}
+
+			// Construct StorageCluster
+			cluster := &corev1.StorageCluster{}
+			cluster.Name = "test-stc"
```

Comment:
we can start define this as a constant, the test case name should be used by all test cases, as some test cases do not uninstall and reinstall stc

Reply:
Done